### PR TITLE
More logging & partition start point

### PR DESCRIPTION
**Pull request by rtgoodwin**
This one is basic so just pulling it over while it can still merge clean. Logging, cleaned up extra readme file, and the proposed change for #80 if that tests out for you.

### CODE REVIEW

File: setup/pi/create-backingfiles-partition.sh [hunk 1]
```diff
@@ -1,8 +1,20 @@
 #!/bin/bash -eu
 
+HEADLESS_SETUP=${HEADLESS_SETUP:-false}
+
+function setup_progress () {
+  SETUP_LOGFILE=/boot/teslausb-headless-setup.log
```
Comment (cimryan) [inline setup/pi/create-backingfiles-partition.sh]: Does this need to be global? If so, I'd like to see it declared and initialized outside the function with the rest of the globals. If not, I'd like to see it declared local.

File: setup/pi/create-backingfiles-partition.sh [hunk 2]
```diff
@@ -15,19 +27,23 @@ LAST_BACKINGFILES_PARTITION_DESIRED_BYTE="$(( $DISK_SIZE - (100 * (2 ** 20)) - 1
 
 ORIGINAL_DISK_IDENTIFIER=$( fdisk -l /dev/mmcblk0 | grep -e "^Disk identifier" | sed "s/Disk identifier: 0x//" )
 
+setup_progress "Modifying partition table for backing files partition..."
 BACKINGFILES_PARTITION_END_SPEC="$(( $LAST_BACKINGFILES_PARTITION_DESIRED_BYTE / 1000000 ))M"
 parted -a optimal -m /dev/mmcblk0 unit B mkpart primary ext4 "$FIRST_BACKINGFILES_PARTITION_BYTE" "$BACKINGFILES_PARTITION_END_SPEC"
 
 LAST_BACKINGFILES_PARTITION_BYTE=$(parted -m /dev/mmcblk0 unit B print | grep -e "^3:" | cut -d ":" -f 3 | sed 's/B//g' )
```
[the remark this answers is not in the document]
Comment (rtgoodwin) [reply setup/pi/create-backingfiles-partition.sh]: I'm of the sense that, in a log, especially a headless one where people may be reporting issues,  more is better. It's how I tracked it down to the particular `parted` giving me trouble. I won't lose sleep over it. :)

File: setup/pi/create-backingfiles-partition.sh (hunk 1 above)
Comment (rtgoodwin) [reply setup/pi/create-backingfiles-partition.sh]: It's resolved for now by putting it local into the function. (Pushed.) 

TBH I'm not 100% adept at tracking the globals, or exporting functions, when multiple script depths come into play at this point, so there's probably a cleaner way to do it all up front. I'd like to explore that with your guidance, and minimize the logic in rc.local even further, but it'll take a bit so hopefully this resolves for now (and gives pointers on what needs to be optimized later).

File: setup/pi/create-backingfiles-partition.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (cimryan) [reply setup/pi/create-backingfiles-partition.sh]: Sorry I wasn't more clear. I was referring to the LAST_BACKINGFILES_PARTITION_BYTE variable.

**Pull request merged**